src/kademlia/dht_tracker.cpp

```diff
@@ -79,14 +79,16 @@ namespace libtorrent { namespace dht
 		, dht_settings const& settings
 		, counters& cnt
 		, dht_storage_interface& storage
-		, dht_state const& state)
+		, dht_state state)
 		: m_counters(cnt)
 		, m_storage(storage)
 		, m_dht(udp::v4(), this, settings, state.nid
 			, observer, cnt, m_nodes, storage)
+		, m_bootstrap_nodes(std::move(state.nodes))
```

Comment:
for the moment yes, but I think it's a good idea to keep the state and provide a method `dht_state::clear()`